webpack.config.ts

```diff
@@ -0,0 +1,367 @@
+        else if (target.StandaloneGeckoView) modifiers.geckoview(manifest)
+        else if (target.Safari) modifiers.safari(manifest)
+        else if (target.E2E) modifiers.E2E(manifest)
+
+        if (env === 'development') modifiers.development(manifest)
+        else modifiers.production(manifest)
+
+        return new ManifestPlugin({ config: { base: manifest } })
+    }
+    function getHotModuleReloadPlugin() {
+        if (!enableHMR) return []
+        // overlay is not working in our environment
+        if (!enableHMRWithReactRefresh) return [new HotModuleReplacementPlugin()]
+        return [new HotModuleReplacementPlugin(), new ReactRefreshWebpackPlugin({ overlay: false })]
+    }
+
+    /** Get environment targets */
+    function getCompilationInfo() {
+        let buildTarget: 'chromium' | 'firefox' | 'safari' | 'E2E' = 'chromium'
+        let firefoxVariant: 'fennec' | 'geckoview' | undefined = undefined
+        let architecture: 'web' | 'app' = 'web'
+        let resolution: 'desktop' | 'mobile' = 'desktop'
+        let buildType: 'stable' | 'beta' | 'insider' = 'stable'
+        if (target.Chromium) buildTarget = 'chromium'
+        if (target.Firefox) buildTarget = 'firefox'
+        if (target.FirefoxForAndroid) firefoxVariant = 'fennec'
+        if (target.StandaloneGeckoView) {
+            firefoxVariant = 'geckoview'
+            architecture = 'app'
+        }
+        if (target.Safari) {
+            buildTarget = 'safari'
+            architecture = 'app'
+        }
+        if (architecture === 'app' || firefoxVariant === 'fennec') resolution = 'mobile'
+        if (target.E2E) buildTarget = 'E2E'
+        if (target.Beta) buildType = 'beta'
+        if (target.Insider) buildType = 'insider'
+
+        // build the envs
+        const allEnv = {
+            STORYBOOK: false,
+            target: buildTarget,
+            build: buildType,
+            architecture,
+            resolution,
+        }
+        if (firefoxVariant) allEnv[firefoxVariant] = firefoxVariant
+        return allEnv
+    }
+    function getSSRPlugin() {
+        if (env === 'development') return []
+        return [new SSRPlugin('popup.html', src('./src/extension/popup-page/index.tsx'))]
+    }
+}
+
+/** All targets available: --firefox --firefox=nightly --firefox-android --firefox-gecko --chromium --wk-webview --e2e */
+function getBuildPresets(argv: any) {
+    return {
+        Firefox: (argv.firefox || argv['firefox-android'] || argv['firefox-gecko']) as 'nightly' | boolean,
+        FirefoxDesktop: argv.firefox as string | boolean,
+        FirefoxForAndroid: !!argv['firefox-android'],
+        StandaloneGeckoView: !!argv['firefox-gecko'],
+        Chromium: !!argv.chromium,
+        Safari: !!argv['wk-webview'],
+        E2E: !!argv.e2e,
+        Beta: !!argv.beta,
+        Insider: !!argv.insider,
+        ReproducibleBuild: !!argv['reproducible-build'],
+    }
+}
+/** Get git info */
+function getGitInfo() {
```

Comment:
i think git info can generate a json file on release (only firefox)

Reply:
good idea, plz open an issue for it, it's beyond of this pr

Reply:
fuck gh-bot issue title checking logic